### PR TITLE
Get All Favorites in Database

### LINKED ISSUE #7: User can get favorites stored in db
When a user sends a `get` request to `/api/v1/favorites` they get a json response with an array of their favorites with `id, title, artistName, genre, rating` in each element.

Example request: `GET /api/v1/favorites` (no params or body required)

Example success response:
```
{ data:
  [
    {
      "id": 1,
      "title": "We Will Rock You",
      "artistName": "Queen"
      "genre": "Rock",
      "rating": 88
    },
    {
      "id": 2,
      "title": "Careless Whisper",
      "artistName": "George Michael"
      "genre": "Pop",
      "rating": 93
    },
  ]
}
```

- [x] Adhere to json api 1.0 and return an object as shown with `{data: []}`
- [x] Sad path for when there are no favorites. Return `status 200` body returns empty array `{data: []}`.
- [x] Testing

### PR DESCRIPTION
**Pull request by rhantak**
- Added `get` action to `/api/v1/favorites`, which responds with a list of all favorites in the database.

Closes #7

### CODE REVIEW

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -49,4 +49,12 @@ router.post('/', (request, response) => {
     .catch((error) => response.status(500).json({error: error}))
 });
 
+router.get('/', (request, response) => {
+  database('favorites')
+    .select('id', 'title', 'artistName', 'genre', 'rating')
+    .then(result => {
+      response.status(200).send({data: result})
+    })
```
Comment (ap2322) [inline routes/api/v1/favorites.js]: Should we have a `.catch` just in case?
```
.catch((error) => response.status(500).json({error: error}))
```

Comment (rhantak) [reply routes/api/v1/favorites.js]: Good 'catch'! I'll add that ASAP

**Comment by rhantak** [comment]
Ok, I've added that catch. Not 100% sure how to test it since I think it's mostly just there for unanticipated things? I think we're done with the card now unless you see anything else that needs fixing.

**Comment by rhantak** [comment]
Just merged in your changes with getting favorites by ID and pulled down to my local branch, all tests are still passing! If you're happy with how this looks it should be production ready.

**Comment by ap2322** [comment]
Thanks for updating the conflicts. I pulled down your branch and ran the tests and it all works. Great work!

**Pull request merged**